### DISCUSSION TITLE
WebFollower, LibreView - Backfill possible?

### DISCUSSION THREAD

**Discussion by schorschl** [NightscoutFoundation/xDrip#3307] [Q&A]
Hi there,

I am using xDrip+ nightly (2db9aa6-2024.01.16 shortly before I used 2023.11.14) but the version isn't the problem.
[Google Pixel 4a, Android 13]

Data source is WebFollower which gets it's data from LibreView / LibreLinkUp. For some reasons unknown, LibreView does not deliver each third value which can be confirmed by looking at the Event Logs (WebFollower:v). Libre Link Up behaves the same. You can set up an alarm in LibreLinkUp in case there hasn't been a reading in the last 5 minutes, which triggers as well. 
So it is clearly an error at the LibreView servers.

My question is: Each web response provides the latest value (which is currently missing each third time) and the history of the last 24h. In this response all values are contained, also the ones that did not report before. Is it possible for xDrip+ to backfill these missing values by parsing the web responses?

![Screenshot_20240123-115419](https://github.com/NightscoutFoundation/xDrip/assets/62949714/f5f6de92-e359-4f8e-8c51-4c1279b25fbb)

**Comment by schorschl** [comment]
Something at LibreView is either buggy concerning timestamps or they are changing the system. The readings in xDrip+ are now saved with a delta of 4-8 minutes instead of each 5 minutes. I am confused. The LibreLinkUp App is also not receiving constant updates and throws errors/alarms.

**Comment by schorschl** [comment]
Actually, xDrip+ is backfilling some measurements which is resulting in the delta of 4-8 minutes between the "dots" which were previously missing. 
BUT the problem is that the LibreView servers are still delivering values with wrong or outdated timestamps.
This causes delays of current values and the total of only 192/288 values per day.

**Comment by arjungm** [comment]
I have a related question to this.

I have my Libre data for the past 7+ months before I started using xdrip+. How do I backfill that data from a CSV?

**Reply by PhilTU** [reply]
to where ?
